RecDP/pyrecdp/primitives/operations/text_perplexity_score.py

```diff
@@ -28,19 +28,43 @@ def __init__(self, language: str = 'en'):
         self.tokenizer = get_model(self.sp_model_key, self.language, 'sentencepiece')
         self.kenlm_model = get_model(self.kl_model_key, self.language, 'kenlm')
 
+    @statistics_decorator
     def process_rayds(self, ds: Dataset) -> Dataset:
         if self.inplace:
             raise NotImplementedError("We don't inplace modify text with normalization")
         else:
             new_name = 'perplexity'
         compute_func = self.get_compute_func()
-        return ds.map(lambda x: self.process_row(x, self.text_key, new_name, compute_func))
+        ret = ds.map(lambda x: self.process_row(x, self.text_key, new_name, compute_func))
+        if self.statistics_flag:
+            self.statistics.max = ret.max(new_name)
```

Comment:
@XinyaoWa , how about we save the result to  self.statistics and only do statistic at the end (summarize function)

Reply:
update for diversity, for others, ray and spark use different operations, still keep in separate functions